### DISCUSSION TITLE
Obsidian plugin?

### DISCUSSION THREAD

**Discussion by alixander** [terrastruct/d2#138] [Ideas]
How much demand is there for this? What would be the features you'd like to see?

**Comment by hcal** [comment]
I literally just came here in a search to find an obsidian plugin

**Reply by alexclaydon** [reply]
I’d love to see this. Obsidian has a lot of traction. I’d be happy if the plugin just allowed Obsidian to correctly render any D2 files in "reading" mode for any given vault, although syntax highlighting like the VS Code extension would also be great.

**Comment by ryanwwest** [comment]
I'd love this.

**Comment by berniexie** [comment]
Obsidian Plugin work in progress!

https://user-images.githubusercontent.com/6413609/204072613-89fb4345-ef8f-4b09-852a-70414b136231.mp4

**Reply by alixander** [reply]
can we keep the previous successful render and append error on top? maybe we could reuse the code in watch that does that

**Comment by alixander** [comment]
this has been submitted to be included in Obsidian's community plugins. https://github.com/terrastruct/d2-obsidian

**Reply by alexclaydon** [reply]
Fantastic - thanks so much for an incredibly quick turnaround. Will try this out this week!

**Comment by alixander** [comment]
This is now in their official list of community plugins: https://obsidian.md/plugins?id=d2-obsidian
